cli/azd/test/functional/initialize_test.go

```diff
@@ -72,15 +72,19 @@ func testCommand(
 	chain []*actions.MiddlewareRegistration,
 	cwd string) {
 	// Run the command when we find a leaf command
-	if testCmd.Runnable() {
+	if testCmd.Runnable() && testCmd.Name() != "azd" {
```

Comment:
I can't help but to feel that we're actively fighting against the library we're using. I wouldn't treat group commands as `Runnable()`, which feels like a bug to me. I don't think there are current implications today, but issues might be more apparent down the road.

I looked at cobra a little, and it seems like the current supported cobra-native way is to do:

1. Add `PersistentRunE()` at `root`.
2. Check if `docs` flag is set.
3. If not set, return `nil` and continue execution.
4. If set,`cmd.SetHelpFunc(<browser impl>)`. Return `flag.ErrHelp`, and the same codepath that handles `--help` will handle `--docs`.

Reply:
If we do want to support this `--docs` flag for all command groups and commands then then @weikanglim 's suggestion is the way to go over middleware.

Reply:
Chatted with @vhvb1989 earlier today, and I missed an important detail in how `Runnable()` in cobra is defined by `RunE`. By the same notion, `PersistentPreRunE` is not invoked if `RunE` is not set. The original proposal doesn't work.

An alternative working implementation here is to add a `pflag.Value` that is aliased as `--help` and calls `SetHelpFunc`:

In cobra_builder.go:

```golang
type docsFlagVal struct {
	help *bool
	c    *cobra.Command
}

func (p *docsFlagVal) Set(s string) error {
	if s != "true" {
		return nil
	}
	*p.help = true

	// set help function to open browser to docs
	p.c.SetHelpFunc(func(c *cobra.Command, _ []string) {
		fmt.Printf("Opened browser to documentation: %s\n", c.CommandPath())
	})
	return nil
}

func (p *docsFlagVal) String() string {
	return fmt.Sprintf("%t", *p.help)
}

func (p *docsFlagVal) Type() string {
	return "bool"
}

// Binds the intersection of cobra command options and action descriptor options
func (cb *CobraBuilder) bindCommand(cmd *cobra.Command, descriptor *actions.ActionDescriptor) error {
	actionName := createActionName(cmd)

	// Automatically adds a consistent help flag
	pHelp := cmd.Flags().BoolP("help", "h", false, fmt.Sprintf("Gets help for %s.", cmd.Name()))

	// docs flags for all commands
	f := docsFlagVal{c: cmd, help: pHelp}
	flag := cmd.Flags().VarPF(
		&f, "docs", "",
		fmt.Sprintf("Opens the documentation for %s in your web browser.", cmd.CommandPath()))
	flag.NoOptDefVal = "true"
```

I do like it a little more than introducing default Runnable logic for all groups and commands, but it could also be viewed as "too hacky" for good taste. If I were to do this, I'd also open an issue for `cobra` to see the appetite in supporting different help-equivalent flags.

Reply:
@weikanglim ,  on this other proposal, parsing `--docs` won't stop the command execution right?
Running, for example `azd up --docs` would open the browser and also run `azd up` 
And if we return an error during parsing the flag, we would need to find out how to tell cobra to do nothing else